### PR TITLE
Updating compass tooltips

### PR DESCRIPTION
**Pull request by lowriech**

### CODE REVIEW

File: src/components/hover.tsx [hunk 1]
```diff
@@ -31,20 +33,27 @@ const Hover: React.FC<HoverProps> = ({
...
     <div
       className={`${classlist.current}`}
-      onMouseEnter={() => setShow(true)}
-      onMouseLeave={() => setShow(false)}
+      onMouseEnter={() => {
+        // setShow(true)
+        action(text);
+      }}
+      onMouseLeave={() => {
+        // setShow(false)
```
Comment (gislawill) [inline src/components/hover.tsx:49]: A little clean up

File: src/components/hover.tsx [hunk 2]
```diff
@@ -31,20 +33,27 @@ const Hover: React.FC<HoverProps> = ({
...
+        action(null);
+      }}
     >
       {children}
-      <div className={"hover" + (show ? " show" : "") + " " + extraClasses}>
+      {/* <div className={"hover" + (show ? " show" : "") + " " + extraClasses}>
         {show ? text : ""}
-      </div>
+      </div> */}
```
Comment (gislawill) [inline src/components/hover.tsx:56]: Can we fully delete this?

File: src/compass/compass.js [hunk 3]
```diff
@@ -22,9 +24,69 @@ function reverseObject(obj) {
...
+      paint: {
+        "fill-color": "#f08",
+        "fill-opacity": 0.4,
+      },
+      opacity: 1,
+      minzoom: 0,
+      maxzoom: 18,
+    };
+    if (!map.getLayer(layerId)) {
+      map.addLayer(layer);
```
Comment (gislawill) [inline src/compass/compass.js:51]: Should this be wrapped in a useEffect? This layer is currently added with every render

**Review by gislawill — CHANGES_REQUESTED** [review]

File: src/compass/compass.js (hunk 3 above)
Comment (lowriech) [reply src/compass/compass.js:51]: I don't think so, it's just in the function definition.  Unless you're talking about the function being redefined every render?

File: src/components/hover.tsx (hunk 1 above)
Comment (lowriech) [reply src/components/hover.tsx:49]: This turns `hover.tsx` into more of a generic action factory.  Not sure if that's a good use case, thoughts?

File: src/compass/compass.js (hunk 3 above)
Comment (gislawill) [reply src/compass/compass.js:51]: Ah you're right, I missed read the parentheses

**Review by gislawill — APPROVED** [review]

**Pull request merged**